src/buffer4/worker.rs

```diff
@@ -153,7 +154,7 @@ where
                 // Using a biased select means the channels will be polled
                 // in priority order, not in a random (fair) order.
                 biased;
-                msg = self.rx1.as_mut().unwrap().recv(), if self.rx1.is_some() => {
+                msg = option_and_then(self.rx1.as_mut(), |rx| rx.recv()) => {
```

Comment:
if you apply my suggestion from below, this could be simplified to remove the closure:
```suggestion
                msg = recv_option(self.rx1.as_mut()) => {
```